### ISSUE TITLE
View bound to method breaks on re-evaluation

### ISSUE THREAD

**Issue by akagomez** [bitovi/canjs#1891]
I couldn't easily find the wording to describe this issue - I think I lack the vocabulary. Fortunately the demo makes it pretty clear.

Actual:

```
Num: 2
Num (x2): function () { return this.attr('num') * 2; }
```

> Not working in latest: http://jsbin.com/zisuxiboga/edit?html,js,output

Expected:

```
Num: 2
Num (x2): 4
```

> Working in 2.2.1: http://jsbin.com/kisapixeve/edit?html,js,output

**Comment by akagomez** [comment]
From what I can tell, the `newVal` starts referring to the compute's method here: https://github.com/bitovi/canjs/blob/master/view/scope/compute_data.js#L102

**Comment by akagomez** [comment]
A bit more context... 

It appears the issue was introduced here: https://github.com/bitovi/canjs/commit/23de6705dd7c97c5608ddb8904541e601a62dd9d 

While addressing an almost identical issue here: https://github.com/bitovi/canjs/issues/1617

**Comment by justinbmeyer** [comment]
Chris said he can port the fix to 2.2.8
